### PR TITLE
feat: reference support

### PR DESCRIPTION
**Pull request by gcandal**

### CODE REVIEW

**Comment by gcandal** [comment]
@joamag

**Comment by joamag** [comment]
@gcandal please change the pending stuff, including the code that is commented out

**Comment by gcandal** [comment]
@joamag

File: js/data/collection.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
 import { NotImplementedError, request } from "../base";
 
+import { Reference, References } from "./typesf";
+
+/**
+ * A mapping from yonius types to the schema types to
+ * be used by the underlying Mongo collection.
+ */
+const MONGO_TYPES = [
+    [Reference, Number],
```
Comment (joamag) [inline js/data/collection.js]: Object ??

File: test/data/model.js [hunk 2]
```diff
@@ -35,6 +35,120 @@ describe("ModelStore", function() {
...
+            await catFriend.save();
+
+            cat.friend = catFriend;
+            await cat.save();
+
+            person.cats = [cat];
+            await person.save();
+
+            person = await mock.Person.get({ id: 1 });
+            assert.strictEqual(person.cats.isResolved, false);
```
Comment (joamag) [inline test/data/model.js:58]: @gcandal add type assertions

**Review by joamag — CHANGES_REQUESTED** [review]

**Comment by gcandal** [comment]
@joamag

File: js/data/model.js [hunk 3]
```diff
@@ -184,8 +248,12 @@ export class Model {
      */
     async _wrap(model) {
         for (const key of Object.keys(this.constructor.schema)) {
-            if (model[key] === undefined) continue;
-            this[key] = model[key];
+            const value = model[key];
+            if (value === undefined) continue;
+            // if (Array.isArray(value) && value.length === 0) {
```
Comment (joamag) [inline js/data/model.js]: remove comments

**Review by joamag — CHANGES_REQUESTED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**